src/machine/machine_k210.go

```diff
@@ -0,0 +1,94 @@
+// +build k210
+
+package machine
+
+import (
+	"device/kendryte"
+	"runtime/interrupt"
+)
+
+func CPUFrequency() uint32 {
+	return 390000000
+}
+
+type PinMode uint8
+
+const (
+	PinInput PinMode = iota
+	PinOutput
+	PinPWM
+	PinSPI
+	PinI2C = PinSPI
+)
+
+// Configure this pin with the given configuration.
+func (p Pin) Configure(config PinConfig) {
+}
+
+// Set the pin to high or low.
+func (p Pin) Set(high bool) {
+}
+
+// Get returns the current value of a GPIO pin.
+func (p Pin) Get() bool {
+	return true
+}
+
+type FPIOA struct {
+	Bus *kendryte.FPIOA_Type
```

Comment:
What is this exactly? It appears to be a kind of I/O multiplexer, similar to what most other MCUs have (like the atsamd chips). But this particular interface is probably very chip specific, which hurts portability. Would it be possible to implement just the GPIO bits here (in the empty `Pin` methods above)?

Reply:
Yes it is very chip specific. I think it is possible to only implement GPIO, I will look into that.